pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1148,18 +1146,32 @@ func buildDefaultCluster(env *model.Environment, name string, discoveryType apiv
 		}
 	}
 
-	defaultTrafficPolicy := buildDefaultTrafficPolicy(env, discoveryType)
+	defaultTrafficPolicy := buildDefaultTrafficPolicy(env, discoveryType, direction, port)
 	applyTrafficPolicy(env, cluster, defaultTrafficPolicy, nil, nil, "",
 		DefaultClusterMode, direction)
 	return cluster
 }
 
-func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Cluster_DiscoveryType) *networking.TrafficPolicy {
+func buildDefaultTrafficPolicy(env *model.Environment, discoveryType apiv2.Cluster_DiscoveryType,
+	direction model.TrafficDirection, port *model.Port) *networking.TrafficPolicy {
 	lbPolicy := DefaultLbType
 	if discoveryType == apiv2.Cluster_ORIGINAL_DST {
 		lbPolicy = networking.LoadBalancerSettings_PASSTHROUGH
 	}
+
+	// Set default circuit breaking for Envoy graceful shutdown
+	var outlierDetectionPolicy *networking.OutlierDetection
+	if direction == model.TrafficDirectionOutbound && (port.Protocol.IsHTTP2() || port.Protocol.IsHTTP()) {
```

Comment:
Will fix. It's HTTP only as the healthcheck filter is an HTTP filter. There's no reason we couldn't set a default circuit breaker for TCP connections, it just wouldn't be affected by failing the healthcheck endpoint.

Reply:
Envoy will not send a new tcp connection request to a ejected endpoint.

Reply:
It will not, but in order for it to be ejected as a result of the upstream Envoy failing its health check, the upstream Envoy's inbound listener must have the HTTP connection manager.

Reply:
>the upstream Envoy's inbound listener must have the HTTP connection manager.

Where it comes from?

Reply:
Talked in private, IC

Reply:
TCP has health checks as well, IIRC, in envoy